src/m365/todo/commands/task/task-remove.ts

```diff
@@ -0,0 +1,151 @@
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
+
+    const getToDoListId = () => {
+      if (args.options.listName) {
+        // Search list by its name
+        const requestOptions: any = {
+          url: `${this.resource}/v1.0/me/todo/lists?$filter=displayName eq '${escape(args.options.listName)}'`,
+          headers: {
+            accept: "application/json;odata.metadata=none"
+          },
+          responseType: 'json'
+        };
+        return request
+          .get(requestOptions)
+          .then((response: any) => response.value && response.value.length === 1 ? response.value[0].id : null);
+      }
+
+      return Promise.resolve(args.options.listId);
+    };
+
+    const removeToDoTask = () => {
+      getToDoListId()
+        .then(toDoListId => {
+          if (!toDoListId) {
+            return Promise.reject(`The list ${args.options.listId || args.options.listName} cannot be found`);
+          }
+
+          const requestOptions: any = {
+            url: `${this.resource}/v1.0/me/todo/lists/${toDoListId}/tasks/${args.options.id}`,
+            headers: {
+              accept: "application/json;odata.metadata=none"
+            },
+            responseType: 'json'
+          };
+
+          return request.delete(requestOptions);
+        })
+        .then((): void => {
+          if (this.verbose) {
+            logger.log(chalk.green('DONE'));
+          }
+
+          cb();
+        }, (err: any) => this.handleRejectedODataJsonPromise(err, logger, cb));
+    };
+
+    if (args.options.confirm) {
+      removeToDoTask();
+    }
+    else {
+      Cli.prompt(
+        {
+          type: "confirm",
+          name: "continue",
+          default: false,
+          message: `Are you sure you want to remove the task ${args.options.id} from task list ${args.options.listId || args.options.listName}?`
+        },
+        (result: { continue: boolean }): void => {
+          if (!result.continue) {
+            cb();
+          }
+          else {
+            removeToDoTask();
+          }
+        }
+      );
+    }
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-i, --id <id>',
+        description: `The id of the task to remove`
+      },
+      {
+        option: '--listName [listName]',
+        description: `The name of the task list in which the task exists. Specify either listId or listName, not both`
+      },
+      {
+        option: '--listId [listId]',
+        description: `The id of the task list in which the task exists. Specify either listId or listName, not both`
+      },
+      {
+        option: '--confirm',
+        description: `Don't prompt for confirmation`
+      },
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(args: CommandArgs): boolean | string {
+
+    if (!args.options.id) {
```

Comment:
In the latest versions of the CLI, we no longer need to manually check for the presence of required options.